lib/core/network/api/user_account_service.dart

```diff
@@ -1,26 +1,49 @@
 import 'package:dio/dio.dart';
-import 'package:hypha_wallet/core/logging/log_helper.dart';
+import 'package:get_it/get_it.dart';
 import 'package:hypha_wallet/core/network/api/endpoints.dart';
+import 'package:hypha_wallet/core/network/api/remote_config_serivice.dart';
 import 'package:hypha_wallet/core/network/networking_manager.dart';
-import 'package:image_picker/image_picker.dart';
 
-// TODO(Nik): these API calls need your help
 class UserAccountService {
   final NetworkingManager networkingManager;
 
   UserAccountService({required this.networkingManager});
 
-  Future<Response> createUserAccount({
+  Future<bool> createUserAccount({
+    required String code,
     required String accountName,
-    required String userName,
-    XFile? image,
+    required String publicKey,
+    required String network,
   }) async {
-    // TODO(Nik): these API calls need your help
-    return await Future.delayed(const Duration(seconds: 3))
-        .then((value) => Response(data: true, requestOptions: RequestOptions(path: 'MOCK')));
-    // return await dioClient.post(Endpoints.userAccountAvailable, data: {
-    //   'userAccount': userAccount,
-    // });
+    final requestBody = '''
+    { 
+      "code": "$code",
+      "accountName": "$accountName",
+      "publicKey": "$publicKey",
+      "network": "$network"
+    }''';
+    final url = GetIt.I<RemoteConfigService>().accountCreatorEndpoint + Endpoints.creteAccount;
```

Comment:
pls inject the service correctly

Reply:
You mean user account service takes a remote config service parameter, right?

Reply:
Ok done - thanks for catching that! 

It's not yet native to me ...